core/src/main/java/org/mskcc/cbio/portal/util/GlobalProperties.java

```diff
@@ -1184,4 +1186,16 @@ public static String getQuerySetsOfGenes() {
         String fileName = portalProperties.getProperty(SETSOFGENES_LOCATION, null);
         return readFile(fileName);
     }
+
+    public static String getWholeSlideViewerToken()
+    {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String secretKey = portalProperties.getProperty(WHOLE_SLIDE_VIEWER_SECRET_KEY);
+        
+        if(authentication != null && authentication.isAuthenticated() && secretKey != null) {
```

Comment:
Is it possible that secretKey is empty?

Reply:
@jjgao I think it will be empty sometime because of this only works for MSK portal, but for the public portal, it may don't exist.

Reply:
then maybe also check `!secretKey.isEmpty()`?